cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -337,7 +315,6 @@ public synchronized void delete(@Name("namespaceId") final NamespaceId namespace
   }
 
   @Override
-  @AuthEnforce(entities = "namespaceId", enforceOn = NamespaceId.class, actions = Action.ADMIN)
```

Comment:
This is delete all the datasets in the namespace, we do not need ADMIN on the namespace, we need ADMIN on all dataset entities.